test/unit_test/fft_test_utils.hpp

```diff
@@ -95,44 +123,57 @@ struct test_params {
     forward_scale = std::get<5>(params);
     backward_scale = std::get<6>(params);
   }
+  explicit test_params(strides_param_tuple params)
+      : test_params(basic_param_tuple{std::get<0>(params), std::get<1>(params), std::get<2>(params),
+                                      std::get<3>(params), std::get<4>(params).lengths}) {
+    strides = std::get<4>(params);
+  }
 };
 
 /// Structure used by GTest to generate the test name
 struct test_params_print {
   std::string operator()(const testing::TestParamInfo<test_params>& info) const {
     auto params = info.param;
     std::stringstream ss;
-    auto print_layout = [&ss](detail::layout layout) {
-      if (layout == detail::layout::PACKED) {
-        ss << "PACKED";
-      } else if (layout == detail::layout::BATCH_INTERLEAVED) {
-        ss << "BATCH_INTERLEAVED";
-      }
-    };
-    auto print_double = [&](double d) {
-      std::string fp_str = std::to_string(d);
-      std::replace(fp_str.begin(), fp_str.end(), '-', 'm');
-      std::replace(fp_str.begin(), fp_str.end(), '.', '_');
-      ss << fp_str;
-    };
 
     ss << "Placement_";
     if (params.placement == placement::IN_PLACE) {
       ss << "IP";
     } else if (params.placement == placement::OUT_OF_PLACE) {
       ss << "OOP";
     }
-    ss << "__LayoutIn_";
-    print_layout(params.input_layout);
-    ss << "__LayoutOut_";
-    print_layout(params.output_layout);
+
+    ss << "__LayoutIn_" << layout_to_string(params.input_layout);
+    ss << "__LayoutOut_" << layout_to_string(params.output_layout);
+
+    if (params.strides) {
+      ss << "__FwdStrides";
+      for (std::size_t s : params.strides.value().forward_strides) {
+        ss << "_" << s;
+      }
+      ss << "__FwdDistance_" << params.strides.value().forward_distance;
+      ss << "__BwdStrides";
+      for (std::size_t s : params.strides.value().backward_strides) {
+        ss << "_" << s;
```

Comment:
We would need to change the test names where ever necessary for now (or probably in every test right away), and omit the LayoutIn and LayoutOut parameters.
For example in the arbitrary strided test suite, mentioning the layouts would not make sense

Reply:
Do you think this needs to be solved now, or when removing the Layout parameters?

Reply:
I would suggest at least in the strided tests, where layout parameters would not make any sense at all, should be omitted in the test name (hence  the "where ever necessary for now"). Further down the line, every test name would specify the layout in terms of stride and distance